File: Backend/app/schemas/backoffice/doctor.py
from ..base import BaseSchema, BaseResponseSchema, add_padded_id
from pydantic import EmailStr, Field
from typing import Optional


class DoctorCreate(BaseSchema):
    department_id: int
    title: Optional[str] = None
    introduction: Optional[str] = None
    # 关联的管理员账号信息（创建医生时同步创建对应的 Admin 账号）
    email: EmailStr
    password: str = Field(..., min_length=8)
    first_name: Optional[str] = None
    last_name: Optional[str] = None


class DoctorUpdate(BaseSchema):
    department_id: Optional[int] = None
    title: Optional[str] = None
    introduction: Optional[str] = None


@add_padded_id()
class DoctorResponse(BaseResponseSchema):
    admin_id: int
    department_id: int
    title: Optional[str] = None
    introduction: Optional[str] = None
    padded_id: Optional[str] = None
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    email: Optional[str] = None
    department_name: Optional[str] = None


class DoctorFilter(BaseSchema):
    department_id: Optional[int] = None
    title: Optional[str] = None

File: medical-admin/src/api/doctor.ts
import request from '../utils/request'
import type { PaginationResponse } from '../types/api'

export interface Doctor {
  id: number
  admin_id: number
  department_id: number
  title?: string
  introduction?: string
  padded_id?: string
  first_name?: string
  last_name?: string
  email?: string
  department_name?: string
}

export interface DoctorCreate {
  department_id: number
  title?: string
  introduction?: string
  email: string
  password: string
  first_name?: string
  last_name?: string
}

export interface DoctorUpdate {
  department_id?: number
  title?: string
  introduction?: string
}

export function listDoctors(page = 1, perPage = 20, departmentId?: number): Promise<PaginationResponse<Doctor>> {
  return request.get('/doctors', { params: { page, per_page: perPage, department_id: departmentId } })
}

export function createDoctor(data: DoctorCreate): Promise<Doctor> {
  return request.post('/doctors', data)
}

export function updateDoctor(id: number, data: DoctorUpdate): Promise<Doctor> {
  return request.put(`/doctors/${id}`, data)
}

export function deleteDoctor(id: number): Promise<void> {
  return request.delete(`/doctors/${id}`)
}
